src/model/enhance-images.ts

```diff
@@ -0,0 +1,333 @@
+import { JSDOM } from 'jsdom';
+
+/**
+ * decideCaption
+ * Decides if a text element should be used to caption the preceding images
+ * @param {TextBlockElement} element - The text element to examine
+ * @param {Boolean} isPhotoEssay - We treat photo essays differently
+ * @param {ImageBlockElement[]} imageBuffer - We use this to decide if a caption is needed or not
+ * */
+const decideCaption = ({
+	element,
+	isPhotoEssay,
+	imageBuffer,
+}: {
+	element?: TextBlockElement;
+	isPhotoEssay: boolean;
+	imageBuffer: ImageBlockElement[];
+}): string => {
+	//  the convention: <ul><li><Caption text</li></ul>
+	if (!element) return '';
+	// We only set a special caption on non photo essay articles if there are more than two
```

Comment:
I didn't really address this point in the PR description. So there's a few reasons but before listing them it's also worth addressing the fact that this change doesn't come without cost. It's more code, adds more maintenance and opens up the surface area for bugs - and by a non trivial amount.

So why do it?

1. The obvious reason is because we can. This type of change would have been very hard in Frontend but is reasonably possible in DCR. Sure, this block of if statements is complex but it doesn't touch any other files 😉 . With this PR + the change in Composer to allow th `halfWidth` role we open up a rich new set of possibilities for how images can be laid out.

2. We remove the needs for workarounds like the one in [this article](https://www.theguardian.com/us-news/2020/dec/11/california-covid-19-coronavirus-surging). At the moment, if editors want to position images side by side then they use this embed. But these workarounds are very broken (wrong font, colours, poor browser support, wrong spacing, etc., etc.) and should have been removed a long time ago. They weren't because there's been no alternative.

3. The original idea came out of a *-rendering meeting where the different platforms were discussing photo essays. Both AR and Editions are yet to add support for photo essays and they are both aware of the large anount of work that is needed to get them to work. Instead, we all agreed that a better solution would be to remove them. As in, to take all the features that make a photo essay a photo essay, and make them universal. Hence this PR.

There's still a ways to go to make that happen and there are still photo essay specific things this PR does not remove but this is a good step toward that larger goal.

Reply:
@nicl @oliverlloyd I think it’s safe to have it supported outside Photo Essays as without Composer changes, this won’t ever appear there. And those will need to be well-thought out before introduction. Having DCR (and AR!) full support everywhere will put a nice pressure on Composer to add support ;-)

Reply:
FWIW, this was also originally mentioned in the [Interactives - Page Modification Contracts](https://docs.google.com/document/d/1qYpabx_PlTWci0V7mPpY2EIvh7xY-e1Wj44m8oDkuEQ/edit#heading=h.f17refnodigk) review as a way to remove one of the ways in which interactives are used to modify the page.